src/main/java/gift/service/ProductService.java

```diff
@@ -104,6 +107,33 @@ public void delete(Long id) {
         productRepository.delete(found);
     }
 
+    @Transactional
+    public ProductOption addOptionTo(Long productId, ProductOptionRequestDto request) {
+        Product product = find(productId);
+        ProductOption created = request.toEntity();
+        product.addOption(created);
+        return created;
+    }
+
+    public List<ProductOptionResponseDto> getOptionsOf(Long productId) {
+        Product product = find(productId);
+        return product.getOptions().stream()
+                .map(ProductOptionResponseDto::from)
+                .toList();
+    }
+
+    @Transactional
+    public void deleteOptionOf(Long productId, Long optionId) {
+        Product product = find(productId);
+        ProductOption option = optionRepository.findById(optionId)
+                .orElseThrow(() -> BusinessException.of(
+                        ResourceErrorCode.PRODUCT_OPTION_NOT_FOUND,
+                        "상품 옵션이 존재하지 않습니다. id = " + optionId,
+                        HttpStatus.NOT_FOUND
+                ));
+        product.removeOption(option);
```

Comment:
옵션이 존재하는지 여부를 optionRepository 에 물어봐야 할까요? 
Product 가 List<ProductOption> 을 가지고 있으니 Product 에게 물어보는건 어떨까요?
아래와 같은 흐름으로 변경이 가능할 것 같은데, 어떻게 생각하시나요?

```suggestion
product.deleteOption(optionId);
//deleteOption
  ProductOption productOption = options 리스트에서 optionId 와 일치하는 옵션 조회. 없으면 예외
  options.remove
  productOption.setProduct(null)
```